src/kani-compiler/cbmc/src/goto_program/stmt.rs

```diff
@@ -176,6 +180,23 @@ impl Stmt {
         stmt!(Assign { lhs, rhs }, loc)
     }
 
+    /// `assert(cond);`
+    pub fn assert_stmt(cond: Expr, prop_class: &str, msg: &str, loc: Location) -> Self {
+        assert!(cond.typ().is_bool());
+
+        // TODO: Extract location metadata in a safer manner
+        let file = loc.filename().unwrap();
+        let line = loc.line().unwrap();
+        let function = loc.function_name();
+        let col = loc.get_column_number().unwrap();
+        let comment = msg.to_string();
+        let property_class = prop_class.to_string();
+
+        let loc = Location::new(file, function, line, Some(col), comment, property_class);
```

Comment:
Why are you destructuring this just to build it up again?

Reply:
So, we need to build a new Location struct object but change just the property_class field. (We are basically setting the field). 

Between adding a setter function and adding the property_class and creating a new object, this seemed to be the safer option as setting even one variable involved making the entire Location struct as mutable. This didn't seem safe to me.

Reply:
I would recommend to remove the getter methods and to use an `if let` expression to extract the fields you need. Something like:

```rust
let new_loc = if let Location::Loc{file, line, function, col, .. } = loc {
        Location::Loc{file, line, function, col, comment: msg.to_string(), property_class: prop_class.to_string());
    } else {
        loc
    }
```